modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/portlet/shared/search/SortPortletSharedSearchContributor.java

```diff
@@ -71,23 +72,33 @@ public void contribute(
 	}
 
 	protected Sort buildSort(String fieldValue, Locale locale) {
-		SortBuilder sortBuilder = _sortBuilderFactory.getSortBuilder();
+		SortOrder sortOrder = SortOrder.ASC;
 
 		if (fieldValue.endsWith("+")) {
-			sortBuilder.field(fieldValue.substring(0, fieldValue.length() - 1));
+			fieldValue = fieldValue.substring(0, fieldValue.length() - 1);
 		}
 		else if (fieldValue.endsWith("-")) {
-			sortBuilder.field(
-				fieldValue.substring(0, fieldValue.length() - 1)
-			).sortOrder(
-				SortOrder.DESC
-			);
+			fieldValue = fieldValue.substring(0, fieldValue.length() - 1);
+			sortOrder = SortOrder.DESC;
 		}
-		else {
-			sortBuilder.field(fieldValue);
+
+		if (fieldValue.startsWith(DDMIndexer.DDM_FIELD_PREFIX)) {
+			try {
+				return ddmIndexer.createDDMStructureFieldSort(
+					fieldValue, locale, sortOrder);
+			}
+			catch (PortalException portalException) {
+				throw new RuntimeException(portalException);
+			}
 		}
 
-		return sortBuilder.locale(
+		SortBuilder sortBuilder = _sortBuilderFactory.getSortBuilder();
+
+		return sortBuilder.field(
+			fieldValue
+		).sortOrder(
```

Comment:
@jorgediaz-lr i think you should SF this in alphabetical order

Reply:
I have added this commit in the new PR [`6e23b01` (#255)](https://github.com/liferay-data-engine/liferay-portal/pull/255/commits/6e23b01f7ca0477746745b428196f1f384979210)